File: 1221. Split a String in Balanced Strings/_1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        l = 0
        r = 0
        res = 0
        for i in s:
            if i == "L":
                l+=1
            else:
                r+=1

            if l == r:
                res+=1
        
        return res

File: 1221. Split a String in Balanced Strings/1221.go
package leetcode

func balancedStringSplit(s string) int {
	l, r, res := 0, 0, 0
	for _,v := range s{
		if v == 'L'{
			l++
		}else{
			r++
		}

		if l==r{
			res++
		}
	}
	return res
}